samples/subsys/lorawan/class_a/src/main.c

```diff
@@ -48,7 +48,7 @@ static void lorwan_datarate_changed(enum lorawan_datarate dr)
 int main(void)
 {
 	const struct device *lora_dev;
-	struct lorawan_join_config join_cfg;
+	struct lorawan_join_config join_cfg = {0};
```

Comment:
There is no need to initialise the join_cfg to 0 here only to set it to 0 later.
I would leave this line as it it